inc/task.class.php

```diff
@@ -648,22 +648,28 @@ function prepareTaskjobs($methods = [], $tasks_id = false) {
 
             // Cancel agents prepared but not in $agent_ids (like computer
             // not in dynamic group)
-            $jobstates_tocancel = $jobstate->find(
-                  ['itemtype' => $item_type,
-                   'items_id' => $item_id,
-                   'plugin_fusioninventory_taskjobs_id' => $job_id,
-                   'NOT'      => [
-                      'state' => [
-                         PluginFusioninventoryTaskjobstate::FINISHED,
-                         PluginFusioninventoryTaskjobstate::IN_ERROR,
-                         PluginFusioninventoryTaskjobstate::CANCELLED,
-                      ]],
-                   'NOT' => [
```

Comment:
Problem is there are tow 'NOT' key in the array  maybe more simple to just move all of them in the same NOT.

Reply:
Or just encapsulate them inside an array. This will produces some extra parenthesis on SQL query, but will do the job.
```
 ['NOT' => [...]],
 ['NOT' => [...]],
```